### DISCUSSION TITLE
First-class functions

### DISCUSSION THREAD

**Discussion by Yethal** [nushell/nushell#5287] [Ideas]
Basically what the title says, it would be really useful to be able to pass functions as arguments to other functions and be able to return them.
Currently it is somewhat possible by using blocks but it's not the same thing (since e.g you can't expect each and every nushell module to export both functions and blocks just so users can pass them around). 
Alternatively something similar to Groovy's .& operator which allows passing references to functions as variables could also work

**Comment by kubouch** [comment]
Yeah, this would be cool. Could you make a feature request issue for it?
